vfs-azure/src/main/java/com/sludev/commons/vfs2/provider/azure/AzFileObject.java

```diff
@@ -512,136 +492,182 @@ protected boolean doSetLastModifiedTime(long modtime) throws Exception {
 
-                            if (fileCurrBlobProperties != null) {
-                                String fileName = srcFile.getName().getBaseName();
-                                String contentType = tika.detect(fileName);
-
-                                log.debug("Content type is {} for {} file", contentType, fileName);
-
-                                if (contentType != null) {
-                                    fileCurrBlobProperties.setContentType(contentType);
-                                }
-                            }
-                            else {
-                                log.debug("currBlobProperties is null");
-                            }
-
-                            OperationContext opContext = new OperationContext();
-                            opContext.setLoggingEnabled(ENABLE_AZURE_STORAGE_LOG);
-
-                            fileCurrBlob.upload(sourceStream, length, null, null, opContext);
-
-                        }
-                        finally {
-                            destFile.close();
-                            srcFile.close();
-                        }
+                        doCopyFromStream(srcFile, destFile);
                     }
                     else {
                         // nothing useful to do if no content and can't have children
                         throw new FileSystemException("vfs.provider/copy-file.error", new Object[] { srcFile, destFile },
                                 new UnsupportedOperationException());
                     }
                 }
-                catch (IOException io) {
-                    throw new FileSystemException("vfs.provider/copy-file.error", new Object[] { srcFile, destFile }, io);
+                catch (IOException | BlobStorageException e) {
+                    throw new FileSystemException("vfs.provider/copy-file.error", new Object[] { srcFile, destFile }, e);
                 }
-                catch (StorageException se) {
-                    throw new FileSystemException("vfs.provider/copy-file.error", new Object[] { srcFile, destFile }, se);
+            }
+        }
+        catch (Exception e) {
+            throw new FileSystemException(e);
+        }
+    }
+
+
+    /**
+     * Copy a file object via input/output streams.
+     *
+     * @param srcFile
+     * @param destFile
+     * @throws IOException
+     */
+    private void doCopyFromStream(FileObject srcFile, FileObject destFile) throws Exception {
+
+        try {
+            String destFilename = destFile.getName().getPath();
+
+            if (destFilename.startsWith("/")) {
+                destFilename = destFilename.substring(1);
+            }
+
+            BlobClient destBlobClient = blobContainerClient.getBlobClient(destFilename);
+
+//                            int blockSize = getBlockSize(srcFile.getContent().getSize(), UPLOAD_BLOCK_SIZE);
+
+            int blockSize = 4 * 1024 *1024;
```

Comment:
this looks like we're hardcoding and ignoring the configurable upload block size... why?